packages/web5/lib/src/dids/did_web/did_web.dart

```diff
@@ -1,49 +1,123 @@
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:web5/src/crypto.dart';
+import 'package:web5/src/dids/bearer_did.dart';
+import 'package:web5/src/dids/did.dart';
 import 'package:web5/src/dids/did_core.dart';
 import 'package:web5/src/dids/did_method_resolver.dart';
-import 'package:web5/src/dids/did.dart';
 
 class DidWeb {
   static const String methodName = 'web';
-  static final resolver = DidMethodResolver(name: methodName, resolve: resolve);
+
+  static final DidMethodResolver resolver = DidMethodResolver(
+    name: methodName,
+    resolve: resolve,
+  );
+
+  static Future<BearerDid> create({
+    required String url,
+    AlgorithmId? algorithm,
+    KeyManager? keyManager,
+    List<String>? alsoKnownAs,
+    List<String>? controllers,
+    List<DidService>? services,
+    List<DidCreateVerificationMethod>? verificationMethods,
+    DidDocumentMetadata? metadata,
+  }) async {
+    algorithm ??= AlgorithmId.ed25519;
+    keyManager ??= InMemoryKeyManager();
+
+    final parsed = Uri.tryParse(url);
+    if (parsed == null) throw 'Unable to parse url $url';
+    final String didId =
+        'did:web:${parsed.host}${parsed.pathSegments.join(':')}';
+
+    final DidDocument doc = DidDocument(
+      id: didId,
+      alsoKnownAs: alsoKnownAs,
+      controller: controllers ?? didId,
+    );
+
+    final List<DidCreateVerificationMethod> defaultMethods = [
+      DidCreateVerificationMethod(
+        algorithm: algorithm,
+        id: '0',
+        type: 'JsonWebKey',
+        controller: didId,
+        purposes: [
+          VerificationPurpose.authentication,
+          VerificationPurpose.assertionMethod,
+          VerificationPurpose.capabilityDelegation,
+          VerificationPurpose.capabilityInvocation,
+        ],
+      ),
+    ];
+
+    final List<DidCreateVerificationMethod> methodsToAdd =
+        verificationMethods ?? defaultMethods;
+
+    for (final DidCreateVerificationMethod vm in methodsToAdd) {
+      final Jwk privateKey = await Crypto.generatePrivateKey(vm.algorithm);
```

Comment:
[`KeyManager.generatePrivateKey`](https://github.com/TBD54566975/web5-dart/blob/main/packages/web5/lib/src/crypto/key_manager.dart#L27) instead of `Crypto`